### PR TITLE
Add LeetCode-1 (different approach) and LeetCode-167 in C++

### LINKED ISSUE #42: Add LeetCode-1. Two SUM, LeetCode-167. Two Sum II
I want to provide a different solution(nlogn) to the problem [Two Sum](https://leetcode.com/problems/two-sum/) and
solution to the problem [Two Sum II](https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/)

### PR DESCRIPTION
**Pull request by r8025n**
resolves #42

### CODE REVIEW

File: Algorithms/Easy/1_TwoSum/SolutionOne.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class Solution {
+public:
+    vector<int> twoSum(vector<int>& nums, int target) {
+        vector<pair<int,int>> valueIndexPair;
```
Comment (Tahanima) [inline Algorithms/Easy/1_TwoSum/SolutionOne.cpp:4]: Change this code to 
```cpp
vector<pair<int, int>> valueIndexPair;
```

File: Algorithms/Easy/1_TwoSum/SolutionOne.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	    vector<int> returnVector;	
+	    int index1, index2, sum;
```
Comment (Tahanima) [inline Algorithms/Easy/1_TwoSum/SolutionOne.cpp:6]: Fix the indentation.

File: Algorithms/Easy/1_TwoSum/SolutionOne.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+        for(int k = 0; k < nums.size(); k++) {
+            valueIndexPair.push_back(pair<int,int>(nums[k], k));
```
Comment (Tahanima) [inline Algorithms/Easy/1_TwoSum/SolutionOne.cpp:9]: Change this code to
```cpp
valueIndexPair.push_back(pair<int, int>(nums[k], k));
```

File: Algorithms/Easy/1_TwoSum/SolutionOne.cpp [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        }
+        
+        sort(valueIndexPair.begin(), valueIndexPair.end());
+        
+	    index1 = 0;
```
Comment (Tahanima) [inline Algorithms/Easy/1_TwoSum/SolutionOne.cpp:14]: Fix indentation

File: Algorithms/Easy/1_TwoSum/SolutionOne.cpp [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	    index2 = valueIndexPair.size() - 1;
```
Comment (Tahanima) [inline Algorithms/Easy/1_TwoSum/SolutionOne.cpp:15]: Fix indentation

File: Algorithms/Easy/1_TwoSum/SolutionOne.cpp [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+        while(index1 < index2) {
+            sum = valueIndexPair[index1].first + valueIndexPair[index2].first;
+            
+	        if(sum == target)
```
Comment (Tahanima) [inline Algorithms/Easy/1_TwoSum/SolutionOne.cpp:20]: Fix indentation

**Review by Tahanima — CHANGES_REQUESTED** [review]
Hi @r8025n 
Please resolve all the requested changes. Also, change the folder name from `167_TwoSum II` to `167_TwoSumII`

**Comment by r8025n** [comment]
HI,
Thank you for your feedback. I have addressed the requested changes, please review.

File: Algorithms/Easy/1_TwoSum/SolutionOne.cpp [hunk 7]
```diff
@@ -0,0 +1,33 @@
+class Solution {
+public:
+    vector<int> twoSum(vector<int>& nums, int target) {
+        vector<pair<int, int>> valueIndexPair;
+	vector<int> answer;
```
Comment (Tahanima) [inline Algorithms/Easy/1_TwoSum/SolutionOne.cpp:5]: Need to fix the indentation.

File: Algorithms/Easy/1_TwoSum/SolutionOne.cpp [hunk 8]
```diff
@@ -0,0 +1,33 @@
+class Solution {
+public:
+    vector<int> twoSum(vector<int>& nums, int target) {
+        vector<pair<int, int>> valueIndexPair;
+	vector<int> answer;	
+	int index1, index2, sum;
```
Comment (Tahanima) [inline Algorithms/Easy/1_TwoSum/SolutionOne.cpp:6]: Need to fix the indentation.

File: Algorithms/Easy/1_TwoSum/SolutionOne.cpp [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+        for(int k = 0; k < nums.size(); k++) {
+            valueIndexPair.push_back(pair<int, int>(nums[k], k));
+        }
+        
+        sort(valueIndexPair.begin(), valueIndexPair.end());
+        
+	index1 = 0;
```
Comment (Tahanima) [inline Algorithms/Easy/1_TwoSum/SolutionOne.cpp:14]: Need to fix the indentation.

File: Algorithms/Easy/1_TwoSum/SolutionOne.cpp [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+	index2 = valueIndexPair.size() - 1;
```
Comment (Tahanima) [inline Algorithms/Easy/1_TwoSum/SolutionOne.cpp:15]: Need to fix the indentation.

File: Algorithms/Easy/1_TwoSum/SolutionOne.cpp [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+            sum = valueIndexPair[index1].first + valueIndexPair[index2].first;
+            
+	    if(sum == target)
+                break;
+            else if(sum < target)
+                index1++;
+            else
+                index2--;
+        }
+
```
Comment (Tahanima) [inline Algorithms/Easy/1_TwoSum/SolutionOne.cpp:27]: Need to fix the indentation of the if-else block.

**Review by Tahanima — CHANGES_REQUESTED** [review]
Hi @r8025n 
Please fix all the indentation issues.

**Comment by r8025n** [comment]
Hi @Tahanima 
I am kind of lost now, could you please help me out  and point me to the right direction? As your recent requested changes are suggesting there are indentation issues in my code but in my forked repo the code looks okay. What am I missing here?

What I did so far:
1. Opened a pull request. 
2. Some changes were requested by you.
3. Addressed the requested changes in my local git repo, updated my repo with the original upstream repo by fetching and merging, then pushed them to my forked repo.
4. And now as you addressed some indentation issues, as I visited my fork repo they seems fine there!!

So did I miss any steps after pushing the first requested changes to my forked repo?

**Review by Tahanima — APPROVED** [review]
@r8025n I get your point. No worries. I'll fix the indentation after merging. Keep up the good work! 😃

**Comment by r8025n** [comment]
Thank You :)

**Pull request merged**